exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/metrics_exporter/__init__.py

```diff
@@ -54,48 +54,54 @@
     MetricsExporter,
     MetricsExportResult,
 )
+from opentelemetry.sdk.metrics.export.aggregate import (
+    HistogramAggregator,
+    LastValueAggregator,
+    MinMaxSumCountAggregator,
+    SumAggregator,
+    ValueObserverAggregator,
+)
 
 logger = logging.getLogger(__name__)
 DataPointT = TypeVar("DataPointT", IntDataPoint, DoubleDataPoint)
 
 
 def _get_data_points(
-    sdk_metric: MetricRecord, data_point_class: Type[DataPointT]
+    sdk_metric_record: MetricRecord, data_point_class: Type[DataPointT]
 ) -> List[DataPointT]:
 
-    data_points = []
-
-    for (
-        label,
-        bound_counter,
-    ) in sdk_metric.instrument.bound_instruments.items():
-
-        string_key_values = []
-
-        for label_key, label_value in label:
-            string_key_values.append(
-                StringKeyValue(key=label_key, value=label_value)
-            )
-
-        for view_data in bound_counter.view_datas:
-
-            if view_data.labels == label:
-
-                data_points.append(
-                    data_point_class(
-                        labels=string_key_values,
-                        value=view_data.aggregator.current,
-                        start_time_unix_nano=(
-                            view_data.aggregator.last_checkpoint_timestamp
-                        ),
-                        time_unix_nano=(
-                            view_data.aggregator.last_update_timestamp
-                        ),
-                    )
-                )
-                break
-
-    return data_points
+    if isinstance(sdk_metric_record.aggregator, SumAggregator):
+        value = sdk_metric_record.aggregator.checkpoint
+
+    elif isinstance(sdk_metric_record.aggregator, MinMaxSumCountAggregator):
+        # FIXME: How are values to be interpreted from this aggregator?
+        raise Exception("MinMaxSumCount aggregator data not supported")
+
+    elif isinstance(sdk_metric_record.aggregator, HistogramAggregator):
+        # FIXME: How are values to be interpreted from this aggregator?
+        raise Exception("Histogram aggregator data not supported")
+
+    elif isinstance(sdk_metric_record.aggregator, LastValueAggregator):
+        value = sdk_metric_record.aggregator.checkpoint
+
+    elif isinstance(sdk_metric_record.aggregator, ValueObserverAggregator):
+        value = sdk_metric_record.aggregator.checkpoint.last
+
+    return [
+        data_point_class(
+            labels=[
+                StringKeyValue(key=str(label_key), value=str(label_value))
+                for label_key, label_value in sdk_metric_record.labels
+            ],
+            value=value,
+            start_time_unix_nano=(
+                sdk_metric_record.aggregator.initial_checkpoint_timestamp
```

Comment:
Is the start time really the first time the aggregator is checkpointed?

Reply:
The start time is not the first time the aggregator is checkpointed, but [_the last time when the aggregation value was reset to "zero"_](https://github.com/open-telemetry/opentelemetry-proto/blob/master/opentelemetry/proto/metrics/v1/metrics.proto#L297).

Reply:
But doesn't this happen in `take_checkpoint`? From the current logic, `self.initial_checkpoint_timestamp = time_ns()` seems to be called when `update` is called.

Reply:
Yes, but it what happens is that the whole metric exporting process ends up a step by calling `take_checkpoint`. If we set the initial checkpoint timestamp to the moment `take_checkpoint` gets called, then we would end up with a timestamp that is greater than the last update.

Reply:
But by definition isn't that fine? Why can't the initial checkpoint timestamp be greater than the last update? If it is used for duration calculations, doesn't it seem that the definition is wrong?

Regardless, even if that isn't an issue, I'm not sure setting the time to the first time an update occurs after checkpointed is correct.

Reply:
Looking at the go implementation, the values for `StartTimeUnixNano` and `TimeUnixNano` appear different for each aggregation, should that not be the case here?

Reply:
Yes, as discussed with @lzchen those changes will be added in future, separate PRs that will hopefully bring this implementation to the level of the Go one.